File: base/src/main/java/com/windscribe/vpn/localdatabase/tables/PopupNotificationTable.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.vpn.localdatabase.tables;


import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@SuppressWarnings("unused")
@Entity(tableName = "notification_table")
public class PopupNotificationTable {

    @PrimaryKey
    @NonNull
    @ColumnInfo(name = "notification_id")
    private final Integer notificationId;

    @ColumnInfo(name = "popup_status")
    private final Integer popUpStatus;

    @ColumnInfo(name = "user_name")
    private String userName;


    public PopupNotificationTable(@NonNull Integer notificationId, String userName, Integer popUpStatus) {
        this.notificationId = notificationId;
        this.userName = userName;
        this.popUpStatus = popUpStatus;
    }

    @NonNull
    public Integer getNotificationId() {
        return notificationId;
    }

    public Integer getPopUpStatus() {
        return popUpStatus;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }
}

File: base/src/main/java/com/windscribe/vpn/localdatabase/tables/PopupNotificationTable.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.vpn.localdatabase.tables

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "notification_table")
class PopupNotificationTable(
    @PrimaryKey
    @ColumnInfo(name = "notification_id")
    val notificationId: Int,
    @ColumnInfo(name = "user_name")
    var userName: String?,
    @ColumnInfo(name = "popup_status")
    val popUpStatus: Int?,
)
